rs/nns/governance/src/neuron_data_validation.rs

```diff
@@ -584,14 +585,18 @@ impl CardinalityAndRangeValidator for KnownNeuronIndexValidator {
     // entry lookup takes ~130K instructions.
     const HEAP_NEURON_RANGE_CHUNK_SIZE: usize = 300000;
     fn validate_cardinalities(neuron_store: &NeuronStore) -> Option<ValidationIssue> {
-        let cardinality_primary_heap = neuron_store
-            .active_neurons_iter()
-            .filter(|neuron| neuron.known_neuron_data.is_some())
-            .count() as u64;
-        let cardinality_primary_stable = with_stable_neuron_store(|stable_neuron_store| {
+        let cardinality_active_neurons = neuron_store.with_active_neurons_iter(|iter| {
+            iter.filter(|neuron| neuron.known_neuron_data.is_some())
+                .count() as u64
+        });
+        let cardinality_stable_neurons = with_stable_neuron_store(|stable_neuron_store| {
             stable_neuron_store.lens().known_neuron_data
         });
-        let cardinality_primary = cardinality_primary_heap + cardinality_primary_stable;
+        let cardinality_primary = if is_active_neurons_in_stable_memory_enabled() {
+            cardinality_stable_neurons
```

Comment:
how will this work during migration?  Do we care?  Will we still need this validator?

Reply:
added a note